CRM/Event/Page/Tab.php

```diff
@@ -182,9 +182,15 @@ public function run() {
   }
 
   public function setContext() {
-    $context = CRM_Utils_Request::retrieve('context',
-      'String', $this, FALSE, 'search'
-    );
+    $urlPath = implode('/', isset($this->urlPath) ? $this->urlPath : NULL);
+    if ($urlPath == 'civicrm/contact/view/participant') {
+      $context = 'participant';
+    }
+    else {
+      $context = CRM_Utils_Request::retrieve('context',
+        'String', $this, FALSE, 'search'
+      );
+    }
```

Comment:
In what scenario would the incoming URL path **not** be `civicrm/contact/view/participant`?